packages/ui-app/src/Params/valueToText.tsx

```diff
@@ -154,7 +154,7 @@ function valueToText (type: string, value: any, swallowError: boolean = true): R
     : div(
       {},
       ['Bytes', 'Data'].includes(type)
-        ? u8aToHex(value.toU8a(true), 512)
+        ? u8aToHex(value.toU8a(true), contentShorten ? 512 : value.length)
```

Comment:
The second part we can keep at `-1` instead of `value.length`

Reply:
That is an interesting one, good to know. With the size of the buffers, the conversion does take some time - with `-1` it should be auto-detected. (here https://github.com/polkadot-js/common/blob/master/packages/util/src/u8a/toHex.ts#L20)

So there _may_ be an underlying bug if they are not acting equivalently or causes lock-ups.

Reply:
Hm it looks more like that if basically bitLength > -1 is given, the Uint8Array is simply 'stringified' by a reducer function rather that anything auto-detected ... Point being that there is a whole lot of integers converted and joined to hex strings which does block the main thread... Looks very costly.. Also I don't really know why Chrome does behave like this.. it feels like that the whole main thread keeps blocked even after 10 minutes and longer and even the DevTools Console is not accessible ..

I tried it with firefox and all went well, also I excluded host implementation problems with ```Uint8Array.prototype.reduce``` or ```Uint8Array.prototype.toString``` on my chrome (version 70.0.3538.110) in case they would not be catched by the transpiler beforehand ..

Also, ```@polkadot/util/u8a/toHex.js u8aToHex()``` has no solution to delegate the workload, it simply doesn't care how large the incoming data actually is. This could be an async operation with a promise returned for example...

Maybe a Service Worker ? :P

Reply:
/ Or use wasm .. ?

Reply:
I'm sure the implementation there can be optimised. Service workers are a bit of an issue, since the utils are environment-agnostic. (Unless we do a specific impl. for the UI itself - which defeats the purpose of having utils.)

Reply:
I thought of an another basic js approach which is to delegate workloads to the event loop.
What we could do is to have a helper which creates a task queue which does create chunks of workloads and delegates them to the event loop (via setTimeout for example). Everything would be handled by promises step by step and the UI would not be blocked when a heavier workload is introduced.

Also there are different approaches, for example this one:
https://github.com/mozilla/task.js